zipline/gens/brokers/rh/robinhood.py

```diff
@@ -0,0 +1,1147 @@
+
+
+class Robinhood:
+    """Wrapper class for fetching/parsing Robinhood endpoints """
+
+    endpoints = {
+        "login": "https://api.robinhood.com/api-token-auth/",
+        "logout": "https://api.robinhood.com/api-token-logout/",
+        "investment_profile": "https://api.robinhood.com/user/investment_profile/",
+        "accounts": "https://api.robinhood.com/accounts/",
+        "ach_iav_auth": "https://api.robinhood.com/ach/iav/auth/",
+        "ach_relationships": "https://api.robinhood.com/ach/relationships/",
+        "ach_transfers": "https://api.robinhood.com/ach/transfers/",
+        "applications": "https://api.robinhood.com/applications/",
+        "dividends": "https://api.robinhood.com/dividends/",
+        "edocuments": "https://api.robinhood.com/documents/",
+        "instruments": "https://api.robinhood.com/instruments/",
+        "margin_upgrades": "https://api.robinhood.com/margin/upgrades/",
+        "markets": "https://api.robinhood.com/markets/",
+        "notifications": "https://api.robinhood.com/notifications/",
+        "orders": "https://api.robinhood.com/orders/",
+        "password_reset": "https://api.robinhood.com/password_reset/request/",
+        "portfolios": "https://api.robinhood.com/portfolios/",
+        "positions": "https://api.robinhood.com/positions/",
+        "quotes": "https://api.robinhood.com/quotes/",
+        "historicals": "https://api.robinhood.com/quotes/historicals/",
+        "document_requests": "https://api.robinhood.com/upload/document_requests/",
+        "user": "https://api.robinhood.com/user/",
+        "watchlists": "https://api.robinhood.com/watchlists/",
+        "news": "https://api.robinhood.com/midlands/news/",
+        "fundamentals": "https://api.robinhood.com/fundamentals/",
+    }
+
+    session = None
+    username = None
+    password = None
+    headers = None
+    auth_token = None
+
+    logger = logging.getLogger('Robinhood')
+    logger.addHandler(logging.NullHandler())
+
+
+    ###########################################################################
+    #                       Logging in and initializing
+    ###########################################################################
+
+    def __init__(self):
+        self.session = requests.session()
+        self.session.proxies = getproxies()
+        self.headers = {
+            "Accept": "*/*",
+            "Accept-Encoding": "gzip, deflate",
+            "Accept-Language": "en;q=1, fr;q=0.9, de;q=0.8, ja;q=0.7, nl;q=0.6, it;q=0.5",
+            "Content-Type": "application/x-www-form-urlencoded; charset=utf-8",
+            "X-Robinhood-API-Version": "1.0.0",
+            "Connection": "keep-alive",
+            "User-Agent": "Robinhood/823 (iPhone; iOS 7.1.2; Scale/2.00)"
+        }
+
+        self.session.headers = self.headers
+
+
+    def login_prompt(self): #pragma: no cover
+        """Prompts user for username and password and calls login() """
+
+        username = input("Username: ")
```

Comment:
I'd really prefer to not to have any interaction. 
If we have interaction it is so much harder to automatically start the app.

The login and password should be specified either via environment variables or application parameters.

Reply:
I make the user info come from a config file so no interaction is required